code/datums/religions.dm

```diff
@@ -6,25 +6,230 @@
 	var/bible_name = "The Holy Bible"
 	var/male_adept = "Chaplain"
 	var/female_adept = "Chaplain"
-	var/bible_type = /obj/item/weapon/storage/bible
+	var/convert_method = "splashing them with holy water, holding a bible in hand."
 
+	var/bible_type = /obj/item/weapon/storage/bible
 	var/obj/item/weapon/storage/bible/holy_book
 
+	var/datum/mind/religiousLeader
+	var/list/datum/mind/adepts = list()
+
 	var/list/bible_names = list()
 	var/list/deity_names = list()
 
+	var/datum/action/renounce/action_renounce
 	var/list/keys = list("christianity") // What you need to type to get this particular relgion.
 
 /datum/religion/New() // For religions with several bibles/deities
 	if (bible_names.len)
 		bible_name = pick(bible_names)
 	if (deity_names.len)
 		deity_name = pick(deity_names)
+	action_renounce = new /datum/action/renounce(src)
+
+/datum/religion/proc/isReligiousLeader(var/mob/living/user)
+	return (user.mind && user.mind == religiousLeader)
 
 // Give the chaplain the basic gear, as well as a few misc effects.
 /datum/religion/proc/equip_chaplain(var/mob/living/carbon/human/H)
 	return TRUE // Nothing to see here, but redefined in some other religions !
 
+/* ---- RELIGIOUS CONVERSION ----
+ * convertAct() -> convertCeremony() -> convertCheck() -> convert()
+ * Redefine 'convertCeremony' to play out your snowflake ceremony/interactions in your religion datum.
+ * In a saner language, convertCeremony() and convertCheck() would be private methods. Those are UNSAFE procs. Call convertAct() instead.
+ */
+
+/* ConvertAct() : here we check if eveything is in place for the conversion, and provide feedback if needed. Sanity for the preacher or the target belongs to the verb in the bible.
+ * - preacher : the guy doing the converting
+ * - subject : the guy being converted
+ * - B : the bible using for the conversion
+ */
+/datum/religion/proc/convertAct(var/mob/living/preacher, var/mob/living/subject, var/obj/item/weapon/storage/bible/B)
+	if (B.my_rel != src) // BLASPHEMY
+		to_chat(preacher, "<span class='warning'>You are a heathen to this God. You feel [B.my_rel.deity_name]'s wrath strike you for this blasphemy.</span>")
+		preacher.fire_stacks += 5
+		preacher.IgniteMob()
+		preacher.emote("scream",,, 1)
+		return FALSE
+	if (preacher != religiousLeader.current)
+		to_chat(preacher, "<span class='warning'>You fail to muster enough mental strength to begin the conversion. Only the Spiritual Guide of [name] can perfom this.</span>")
+		return FALSE
+	if (subject.mind.faith == src)
+		to_chat(preacher, "<span class='warning'>You and your target follow the same faith.</span>")
+		return FALSE
+	if (subject.mind.faith != src && subject.mind.faith.isReligiousLeader(subject))
```

Comment:
The first check will never be false because of the check on line 58, the second check will runtime if `faith `is null

Reply:
I'll just replace it with a `istype()`